### PR TITLE
Added fallback when external IP returned by router is a private network IP

### PR DESCRIPTION
**Pull request by Measurity**
Subnet calculator: https://jodies.de/ipcalc

### CODE REVIEW

File: Nitrox.Test/Model/Helper/NetHelperTest.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using System.Net;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using NitroxModel.Helper;
+
+namespace Nitrox.Test.Model.Helper
```
Comment (Jannify) [inline Nitrox.Test/Model/Helper/NetHelperTest.cs:6]: Would give it the same namespace as NetHelper.

File: Nitrox.Test/Model/Helper/NetHelperTest.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+{
+    [TestClass]
+    public class NetHelperTest
+    {
+        [TestMethod]
+        public void ShouldMatchPrivateIps()
+        {
+            IPAddress.Parse("192.168.2.1").IsPrivate().Should().BeTrue();
```
Comment (Jannify) [inline Nitrox.Test/Model/Helper/NetHelperTest.cs:14]: Shouldn't we test all the ips (or just the start and end of the ranges) defined in NetHelper? Why only a few manually selected?

Comment (Measurity) [reply Nitrox.Test/Model/Helper/NetHelperTest.cs:14]: Just laziness. Will add the others when I can.

**Review by Jannify — APPROVED** [review]

File: NitroxModel/Helper/NetHelper.cs [hunk 3]
```diff
@@ -13,6 +15,17 @@ namespace NitroxModel.Helper
 {
     public static class NetHelper
     {
+        private static readonly ReadOnlyCollection<string> privateNetworks =
```
Comment (dartasen) [inline NitroxModel/Helper/NetHelper.cs:18]: Do we really need ReadOnlyCollection wrapper here ?

File: NitroxModel/Helper/NetHelper.cs [hunk 4]
```diff
@@ -124,5 +137,31 @@ public static IPAddress GetHamachiIp()
...
+            {
+                string[] parts = mask.Split('/');
+
+                int ipNum = BitConverter.ToInt32(ipAddress.GetAddressBytes(), 0);
+                int cidrAddress = BitConverter.ToInt32(IPAddress.Parse(parts[0]).GetAddressBytes(), 0);
+                int cidrMask = IPAddress.HostToNetworkOrder(-1 << (32 - int.Parse(parts[1])));
+
+                return (ipNum & cidrMask) == (cidrAddress & cidrMask);
+            }
+
```
Comment (dartasen) [inline NitroxModel/Helper/NetHelper.cs:156]: Maybe we can check if the adress.IsIPv4MappedToIpv6 to convert the adress to a standard IPV4 using MapToIpv4() method, will allow us to cover more case

**Review by tornac1234 — APPROVED** [review]

File: NitroxModel/Helper/NetHelper.cs (hunk 4 above)
Comment (Measurity) [reply NitroxModel/Helper/NetHelper.cs:156]: This seems too niche to be worth adding at this moment.

**Review by dartasen — APPROVED** [review]

**Pull request merged**